server/src/internalClusterTest/java/org/elasticsearch/health/HealthMetadataServiceIT.java

```diff
@@ -147,9 +163,20 @@ public void testWatermarkSettingUpdate() throws Exception {
                 var healthMetadata = HealthMetadata.getFromClusterState(internalCluster.clusterService().state());
                 var diskMetadata = healthMetadata.getDiskMetadata();
                 assertThat(diskMetadata.describeHighWatermark(), equalTo(updatedHighWatermark));
-                assertThat(diskMetadata.highMaxHeadroom(), equalTo(updatedHighMaxHeadroom));
+
+                // The value of the setting `cluster.routing.allocation.disk.watermark.high.max_headroom` depends upon the existence of
+                // `cluster.routing.allocation.disk.watermark.high`.
+                // Check {@link DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_MAX_HEADROOM_SETTING}
+                assertThat(diskMetadata.highMaxHeadroom(), equalTo(percentageMode ? updatedHighMaxHeadroom : ByteSizeValue.ofGb(150)));
                 assertThat(diskMetadata.describeFloodStageWatermark(), equalTo(updatedFloodStageWatermark));
-                assertThat(diskMetadata.floodStageMaxHeadroom(), equalTo(updatedFloodStageMaxHeadroom));
```

Comment:
Is this flood headroom check removal on purpose?

Reply:
nop, it's a couple of lines bellow :P  (176-179)

Reply:
same comment that https://github.com/elastic/elasticsearch/pull/96843/files#r1230675689

Reply:
Ah thanks 👍 🚀